Parsers/ASimAuthentication/Parsers/ASimAuthenticationCiscoMeraki.yaml

```diff
@@ -0,0 +1,103 @@
+      "5","Other",
+      "6","Other",
+      "7","Other",
+      "8","Other",
+      "9","Other",
+      "10","Logon violates policy",
+      "11","Logon violates policy",
+      "12","Other",
+      "13","Logon violates policy",
+      "14","Other",
+      "15","Other",
+      "16","Other",
+      "17","Other",
+      "18","Incorrect key",
+      "19","Incorrect key",
+      "20","Incorrect key",
+      "21","Other",
+      "22","Other",
+      "23","Other",
+      "24","Logon violates policy",
+  ];
+  let parser = (disabled:bool=false){
+    union isfuzzy=true
+          (
+        meraki_CL
+        | project-rename LogMessage =  Message
+          ),
+          (
+        Syslog
+        | project-rename LogMessage =  SyslogMessage
+          )
+      | where not(disabled) and LogMessage has "events" and (LogMessage has_any ("8021x_auth", "wpa_auth", "splash_auth", "8021x_deauth", "8021x_client_deauth", "wpa_deauth", "8021x_eap_failure", "8021x_eap_success") or (LogMessage has "disassociation" and LogMessage has "auth_neg_failed"))
+      | extend Parser = extract_all(@"(\d+.\d+)\s([\w\-\_]+)\s([\w\-\_]+)\s([\S\s]+)$", dynamic([1, 2, 3, 4]), LogMessage)[0]
+      | extend Epoch = tostring(Parser[0]),
+              Device = tostring(Parser[1]),
+              Substring = tostring(Parser[3]),
+              EventCount=int(1),
+              EventProduct="Meraki",
+              EventVendor="Cisco",
+              EventSchema="Authentication",
+              EventSchemaVersion="0.1.3"
+      | extend EpochTimestamp = split(Epoch,".")
+      | extend EventStartTime = unixtime_seconds_todatetime(tolong(EpochTimestamp[0])),
+              EventEndTime=TimeGenerated
+      | extend TypeOfEvent = case(
+                                Substring has_any ("8021x_auth", "wpa_auth", "splash_auth", "8021x_eap_success"), "Authentication",
+                                Substring has_any ("8021x_deauth", "8021x_client_deauth", "wpa_deauth"), "DeAuthentication",
+                                Substring has_any ("8021x_eap_failure", "disassociation"), "AuthenticationFailure",
+                                ""
+                        )
+      | invoke _ASIM_ResolveDvcFQDN('Device')
+      | extend Dvc = DvcHostname,
+              EventResult = case(
+                            TypeOfEvent == "Authentication" or TypeOfEvent == "DeAuthentication", "Success",
+                            TypeOfEvent == "AuthenticationFailure", "Failure",
+                            ""
+                        ),
+              EventType = case(
+                              TypeOfEvent == "Authentication" or TypeOfEvent == "AuthenticationFailure", "Logon",
+                              TypeOfEvent == "DeAuthentication", "Logoff",
+                            ""
+                        ),
+              DvcIpAddr = extract(@"(src|last_known_client_ip|ip_src|ip|client_ip)[ ]{0,1}=[ ]{0,1}\'*((?:[0-9]{1,3}\.){3}[0-9]{1,3})",2,Substring),
```

Comment:
Extract all the fields using parse-kv with pair_delimiter as " "(space) and key-value delimiter as "=". Further, use coalesce to set the value as per availability of the field.
Fields where this can be useful is DvcIpAddr, DvcMacAddr, User, AdditionalFields, reason

Reply:
Done where ever applicable